src/tss2-esys/api/Esys_LoadExternal.c

```diff
@@ -163,6 +167,23 @@ Esys_LoadExternal_Async(
         return r;
     esysContext->state = _ESYS_STATE_INTERNALERROR;
 
+    /* Pad ECC parameters */
+    if (inPublic && inPublic->publicArea.type == TPM2_ALG_ECC) {
+        r = ecc_get_keysize(inPublic->publicArea.parameters.eccDetail.curveID, &keysize);
+        if (r != TSS2_RC_SUCCESS)
+            return r;
+
+        inPublic_padded = *inPublic;
+        ecc_public_add_padding(&inPublic_padded, keysize);
+        inPublic = (const TPM2B_PUBLIC*) &inPublic_padded;
```

Comment:
This will mess up the HMAC sessions. What if someone would want to precompute the auth of a session, where he sends a command and one of the parameters is the object handle of the just loaded ECC key. 
The name computed for the the original inPublic.public will not be valid as soon as the Esys_LoadExternal_Async() returns. What will happen if one would then bound a new session to this object? Again - in my opinion the padding should be done by the user of the API.

Reply:
You're totally right. I did not notice this issue until now. Padding the public portion of the key changes its name from name<sub>unpadded</sub> to name<sub>padded</sub>. This might be a problem if the user assumes name<sub>unpadded</sub> because it is used to calculate authHMAC for the HMAC session.

Sorry for the inconvenience.